http-client/src/main/java/io/airlift/http/client/ResponseHandler.java

```diff
@@ -25,4 +28,27 @@ T handleException(Request request, Exception exception)
 
     T handle(Request request, Response response)
             throws E;
+
+    interface BaseResponse<T>
```

Comment:
This interface seems rather specific to put in `ResponseHandler`. Maybe a top level `ValueResponse`?